presto-main/src/main/java/com/facebook/presto/operator/JoinProbe.java

```diff
@@ -15,20 +15,124 @@
 
 import com.facebook.presto.spi.Page;
 import com.facebook.presto.spi.PageBuilder;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.Type;
 
-public interface JoinProbe
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+
+public class JoinProbe
```

Comment:
If you separate interface removal and class renaming into two commits, it will be easier to review.

Reply:
Ya I was thinking about the same; but that requires changes to all references of `JoinProbe` to `SimpleJoinProbe` then back to `JoinProbe` again in two commits. (Same for `LookupJoinPageBuilder`/`DictionaryLookupJoinPageBuilder` change as well.)